src/daft-logical-plan/src/optimization/rules/push_down_projection.rs

```diff
@@ -544,6 +544,10 @@ impl PushDownProjection {
                 // Cannot push down past a Pivot/MonotonicallyIncreasingId because it changes the schema.
                 Ok(Transformed::no(plan))
             }
+            LogicalPlan::Window(_) => {
+                // Cannot push down past a Window because it changes the window calculation results
+                Ok(Transformed::no(plan))
+            }
```

Comment:
We might be able to push a projection past a window if the projection keeps all the columns necessary for window. Either way it's ok to not optimize it for now